docs/source/ser_deser.rst

```diff
@@ -0,0 +1,78 @@
+Serialization and Deserialization for torchao optimized models
+--------------------------------------------------------------
+
+Serialization and deserialization is an important question that people care about especially when we integrate torchao with other libraries. Here we want to describe how serialization and deserialization works for torchao optimized (quantized or sparsified) models.
+
+High level serialization and deserialization flow
+=================================================
+
+```python
+import copy
+import tempfile
+import torch
+from torchao.quantization.quant_api import (
+    quantize_,
+    int4_weight_only,
+)
+
+class ToyLinearModel(torch.nn.Module):
+    def __init__(self, m=64, n=32, k=64):
+        super().__init__()
+        self.linear1 = torch.nn.Linear(m, n, bias=False)
+        self.linear2 = torch.nn.Linear(n, k, bias=False)
+
+    def example_inputs(self, batch_size=1, dtype=torch.float32, device="cpu"):
+        return (torch.randn(batch_size, self.linear1.in_features, dtype=dtype, device=device),)
+
+    def forward(self, x):
+        x = self.linear1(x)
+        x = self.linear2(x)
+        return x
+
+dtype = torch.bfloat16
+m = ToyLinearModel(1024, 1024, 1024).eval().to(dtype).to("cuda")
+example_inputs = m.example_inputs(dtype=dtype, device="cuda")
+quantize_(m, int4_weight_only())
+ref = m(*example_inputs)
+with tempfile.NamedTemporaryFile() as f:
+    torch.save(m.state_dict(), f)
+    f.seek(0)
+    state_dict = torch.load(f)
+
+with torch.device("meta"):
+    m_loaded = ToyLinearModel(1024, 1024, 1024).eval().to(dtype)
+    
+m_loaded.load_state_dict(state_dict, assign=True)
+res = m_loaded(*example_inputs)
+assert torch.equal(res, ref)
+```
+
+What happens when serializing an optimized model?
+=================================================
+To serilize an optimized model, we just need to call `torch.save(m.state_dict(), f)`, because in torchao, we use tensor subclass to represent different dtypes or support different optimization techniques like quantization and sparsity. So after optimization, the only thing that is updated is the weight Tensor and the model structure is not changed at all. For example:
```

Comment:
The subclass point is not well explained, I think what you're trying to say more plainly is at model save/load time we swap in the quantized weights 

Which means we're instantiating the full precision model, which means we probably also want to explain why people might want to instantiate a model on cpu to later transfer to gpu

Reply:
we recommend people to initialize a model in meta device, this is explained in the deserialization section.

Reply:
does the example help with the explanation? otherwise let me know what else I can change